src/persist/src/location.rs

```diff
@@ -1035,6 +1046,32 @@ pub mod tests {
             Ok(Ok(()))
         );
 
+        // Sequence numbers used within Consensus have to be within [0, i64::MAX].
+        let sequence_number_edge_cases = vec![
+            ("zero".to_string(), 0, true),
+            (
+                "i64_max".to_string(),
+                i64::MAX.try_into().expect("i64::MAX fits in u64"),
+                true,
+            ),
+            ("i64_max_plus_1".to_string(), 1 << 63, false),
+            ("u64_max".to_string(), u64::MAX, false),
+        ];
+
+        for (key, seqno, expected_success) in sequence_number_edge_cases {
+            let data = VersionedData {
+                seqno: SeqNo(seqno),
+                data: "data".as_bytes().to_vec(),
+            };
+
+            let result = consensus.compare_and_set(&key, deadline, None, data).await;
```

Comment:
this is better expressed as just asserting on the compare_and_set call with things inlined. as-is, the "table" based testing doesn't give you much indication about which case failed. you could get around this by adding more contexts to the asserts but in this case, it's really not too much boilerplate to just assert directly

```rust
assert_eq!(consensus.compare_and_set(&"zero", deadline, None, VersionedData {
    seqno: SeqNo(0),
    data: vec![],
}).await, Ok(()));
```

(or however rustfmt deals with that line)

Reply:
ok will do!